osscan2.cc

```diff
@@ -3411,7 +3411,7 @@ bool HostOsScan::get_tcpopt_string(struct tcp_hdr *tcp, int mss, char *result, i
       memcpy(&tmpshort, q, 2);
       /*  if (ntohs(tmpshort) == mss) */
       /*    *p++ = 'E'; */
-      sprintf(p, "%hX", ntohs(tmpshort));
+      sprintf(p, "%hhX", ntohs(tmpshort));
```

Comment:
This is incorrect: `hhX` is an `unsigned char`, but `ntohs` returns a `uint16_t`. What warning was given for `hX` in this context?